src/llvm-final-gc-lowering.cpp

```diff
@@ -314,55 +237,46 @@ bool FinalLowerGC::runOnFunction(Function &F)
-    auto queueGCRootFunc = getOrNull(jl_intrinsics::queueGCRoot);
-    auto safepointFunc = getOrNull(jl_intrinsics::safepoint);
+    queueRootFunc = getOrDeclare(jl_well_known::GCQueueRoot);
+    poolAllocFunc = getOrDeclare(jl_well_known::GCPoolAlloc);
+    bigAllocFunc = getOrDeclare(jl_well_known::GCBigAlloc);
+    allocTypedFunc = getOrDeclare(jl_well_known::GCAllocTyped);
+    T_size = F.getParent()->getDataLayout().getIntPtrType(F.getContext());
+
+    DenseMap<Value *, void(FinalLowerGC::*)(CallInst *, Function &)> intrinsic_map;
+#define LOWER_INTRINSIC(INTRINSIC, LOWER_INTRINSIC_FUNC) \
+    do { \
+        auto intrinsic = getOrNull(jl_intrinsics::INTRINSIC); \
+        if (intrinsic) { \
+            intrinsic_map[intrinsic] = &FinalLowerGC::LOWER_INTRINSIC_FUNC; \
+        } \
+    } while (0)
+
+    LOWER_INTRINSIC(newGCFrame, lowerNewGCFrame);
+    LOWER_INTRINSIC(pushGCFrame, lowerPushGCFrame);
+    LOWER_INTRINSIC(popGCFrame, lowerPopGCFrame);
+    LOWER_INTRINSIC(getGCFrameSlot, lowerGetGCFrameSlot);
+    LOWER_INTRINSIC(GCAllocBytes, lowerGCAllocBytes);
+    LOWER_INTRINSIC(queueGCRoot, lowerQueueGCRoot);
+    LOWER_INTRINSIC(safepoint, lowerSafepoint);
+
+#undef LOWER_INTRINSIC
 
     // Lower all calls to supported intrinsics.
-    for (BasicBlock &BB : F) {
-        for (auto it = BB.begin(); it != BB.end();) {
-            auto *CI = dyn_cast<CallInst>(&*it);
-            if (!CI) {
-                ++it;
+    for (auto &BB : F) {
+        for (auto &I : make_early_inc_range(BB)) {
+            auto *CI = dyn_cast<CallInst>(&I);
+            if (!CI)
                 continue;
-            }
 
             Value *callee = CI->getCalledOperand();
             assert(callee);
-
-            if (callee == newGCFrameFunc) {
-                replaceInstruction(CI, lowerNewGCFrame(CI, F), it);
-            }
-            else if (callee == pushGCFrameFunc) {
-                lowerPushGCFrame(CI, F);
-                it = CI->eraseFromParent();
-            }
-            else if (callee == popGCFrameFunc) {
-                lowerPopGCFrame(CI, F);
-                it = CI->eraseFromParent();
-            }
-            else if (callee == getGCFrameSlotFunc) {
-                replaceInstruction(CI, lowerGetGCFrameSlot(CI, F), it);
-            }
-            else if (callee == GCAllocBytesFunc) {
-                replaceInstruction(CI, lowerGCAllocBytes(CI, F), it);
-            }
-            else if (callee == queueGCRootFunc) {
-                replaceInstruction(CI, lowerQueueGCRoot(CI, F), it);
-            }
-            else if (callee == safepointFunc) {
-                lowerSafepoint(CI, F);
-                it = CI->eraseFromParent();
-            }
-            else {
-                ++it;
-            }
+            
+            auto it = intrinsic_map.find(callee);
+            if (it == intrinsic_map.end())
+                continue;
+            
+            (this->*(it->second))(CI, F);
```

Comment:
This seems like it might be simpler and very slightly faster by putting the branch part of the `LOWER_INTRINSIC` macro here, as it was before and repeat the list, but anyways, that is just a minor style difference. The PR LGTM as-is.